src/test_creation/modules/workflow/parse.py

```diff
@@ -7,28 +7,51 @@
 from .response import EvaluationResponse
 from ..mixins import ExportableMixin
 
+from modules.code_analyzer.repo import Repository
+
 
 class ResponseParser(ExportableMixin):
-    def __init__(self, response: EvaluationResponse):
+    def __init__(self, response: EvaluationResponse, respository: Repository = None):
+        # FIXME: respository is required to extract the line numbers for functions
+        #        I added an optional argument "respository" here, can't think of any better way to handle it yet
         self.response = response
         self.evaluation_report = None
+        self.repository = respository
+        self.items = []
+
+    def _parse_items(self):
+        items = []
+        for result in self.response.call_results:
+            response = result.parsed_response['results']
+            for item in response:
+                fp = result.files_evaluated[0] # FIXME: it might fail if the evaluation is on multiple files
```

Comment:
I think it is good enough for now... as long as we add `FIXME`/`TODO` to remind ourselves haha